Error-reporting command

rosen-vladimirov opened the pull request:
Add error-reporting command that should be used in a similar manner as feature-usage-tracking command. The difference is that during installation, the exceptions tracking will be enabled without prompt (keep current behavior that we are tracking exceptions). It can be disabled later by using `$ appbuilder error-reporting disable`.
Refactor analytics-service in order to "hide" some methods from the interface and to simplify adding of new analytics commands and variables. 

**Rename feature-usage-tracking to usage-reporting**

rosen-vladimirov commented on services/analytics-service.ts:
```diff
@@ -21,13 +21,16 @@ export class AnalyticsService implements IAnalyticsService {
-		private $options: IOptions) { }
+		private $options: IOptions) {
+			this.settingsNames = [this.$staticConfig.TRACK_FEATURE_USAGE_SETTING_NAME, this.$staticConfig.ERROR_REPORT_SETTING_NAME];
+			this.settingsNames.forEach(settingName => this.getStatus(settingName).wait());
```
This should not be here as it is causing cyclic dependencies error in some cases

teobugslayer commented on commands/analytics.ts:
```diff
@@ -6,23 +6,64 @@ import util = require("util");
+					this.$logger.out(this.$analyticsService.getStatusMessage(settingName, this.$options.json, "Usage reporting").wait());
+					break;
+				default:
+					this.$errors.fail(util.format("The value '%s' is not valid. Valid values are 'enable', 'disable' and 'status'.", arg));
```
Consider using template strings instead.

teobugslayer commented on commands/analytics.ts:
```diff
@@ -32,18 +73,19 @@ export class AnalyticsCommand implements ICommand {
 				case "disable":
 				case "status":
-				case undefined:
+				case "":
```
We can use template string on line 91 :)

rosen-vladimirov replied:
done :)

teobugslayer commented on services/analytics-service.ts:
```diff
@@ -186,70 +183,75 @@ export class AnalyticsService implements IAnalyticsService {
+	private isEverythingDisabled(): boolean {
+		let statuses = _(this.analyticsStatuses)
+						.values()
+						.groupBy(p => p)
```
or perhaps ```.groupBy(_.identity) ```

Fatme commented on commands/analytics.ts:
```diff
@@ -1,52 +1,88 @@
+		return(() => {
+			let arg = args[0] || "";
+			let settingName = this.$staticConfig.ERROR_REPORT_SETTING_NAME;
+			switch(arg.toLowerCase()) {
```
This code is duplicated with the code in AnalyticsCommand. Consider reusing it.

rosen-vladimirov commented:
ping @Fatme  - I've changed the commands as you have suggested in order to reuse the code, could you take a look
@teobugslayer I've fixed the code based on your comments

teobugslayer commented:
:+1:
